exotica_python/src/pyexotica.cpp

```diff
@@ -960,8 +961,17 @@ PYBIND11_MODULE(_pyexotica, module)
     scene.def("get_model_joint_names", &Scene::GetModelJointNames);
     scene.def("get_model_state", &Scene::GetModelState);
     scene.def("get_model_state_map", &Scene::GetModelStateMap);
+    scene.def("get_tree_map", [](Scene& scene) {
+        std::map<std::string, std::shared_ptr<KinematicElement>> map_shared;
+        for (const auto& m : scene.GetTreeMap())
+        {
+            map_shared[m.first] = m.second.lock();
+        }
+        return map_shared;
+    });
     scene.def("set_model_state", (void (Scene::*)(Eigen::VectorXdRefConst, double, bool)) & Scene::SetModelState, py::arg("x"), py::arg("t") = 0.0, py::arg("update_trajectory") = false);
     scene.def("set_model_state_map", (void (Scene::*)(std::map<std::string, double>, double, bool)) & Scene::SetModelState, py::arg("x"), py::arg("t") = 0.0, py::arg("update_trajectory") = false);
+    scene.def("set_frame_state", &Scene::SetFrameState);
```

Comment:
`attach_object_local ` already implements this.

Reply:
`Scene::AttachObject` and `Scene::AttachObjectLocal` seem to do much more.
`Scene::SetFrameState` was intended to just change the pose of a frame, that is already part of the scene.

With `AttachObject` the behaviour will be inconsistent:
If I add a frame via the scene initialiser, it will not be part of the `attached_objects_`. A `DetachObject` will raise an exception in this case. Once I change the pose with `AttachObject`, a `DetachObject` call will not throw. Also, this will not really detach the object, but attach it with the identity transform to the root frame.

Reply:
What is the usage of `attached_objects_` actually? Is it to keep track of the object's pose when updating frames?

In:
https://github.com/ipab-slmc/exotica/blob/59fb6585561f2f5f07930b9a6aa789a1cb2dbd2b/exotica_core/src/scene.cpp#L546-L549
it appears as if all attached objects get attached locally again, with the same name, parent and pose.

Reply:
The attached objects keep track of objects attached to the robot after resetting/reloading the scene. After the scene is re-loaded, the objects gets re-attached to the same objects they were attached to before re-loading. Since you may not have the parent object created yet, you don't want to just blindly attach objects to parents that may be added in the next iteration. It is anly required when you change the parent.

`AttachObjectLocal` does a little extra work because it allows you to change the parent.
It makes the setting of the frame explicit, you always specify the parent by name so that frame you provide is in the parent frame. Unless the time to look up the new parent is too much for you, just use `AttachObjectLocal`. It calls the the same function to update the to-tip-frame.

Reply:
Ok, I am using `AttachObjectLocal` now and removed `SetFrameState`.
The scene initiliser has a `AttachLinks` that I can use in place of `Links` to circumvent the inconsistency with `DetachObject`.